packages/monaco/src/browser/monaco-outline-contribution.ts

```diff
@@ -342,6 +347,7 @@ export interface MonacoOutlineSymbolInformationNode extends OutlineSymbolInforma
     uri: URI;
     range: Range;
     fullRange: Range;
+    detail: string;
```

Comment:
I have not yet tried your changes in action but I think this should be `detail?: string` instead. See the [`DocumentSymbol.detail`](https://microsoft.github.io/language-server-protocol/specification#textDocument_documentSymbol) in the specification.

Reply:
I thought it should be `detail?: string`, but then I found its definition in `monaco.d.ts` in `node_modules/@typefox/monaco-editor-core/` which tells that it's not optional:

![documentsymbol detail](https://user-images.githubusercontent.com/620781/48950767-957eee00-ef3c-11e8-9197-328a71f1722d.png)

That's why I've made it to be `detail: string`. Still not sure if it's correct, but changing it to be optional will change nothing as it will be always initialized from `DocumentSymbol.detail` which is not an optional.

Reply:
> `DocumentSymbol.detail` which is not an optional.

@akosyakov, what's the truth? The TS-LS fills the `detail` with an empty string, but according to the LSP, `DocumentSymbol.detail` is optional. Thanks!

Reply:
Since we don't control monace.d.ts we will have to handle the discrepancy: note that ```null``` is a valid value for a non-optional field, so we just have to handle the case of "undefined field" in the LSP version of DcocumentSymbol correctly.

Reply:
> Since we don't control monace.d.ts

FYI: [`@typefox/monaco-editor-core`](https://www.npmjs.com/package/@typefox/monaco-editor-core)

Reply:
What I mean is that it doesn't really matter: we either have a value or we don't. If monaco wants a value, we can pass null if we don't have on.

Reply:
@akosyakov, ping.

Reply:
> what's the truth?

In VS Code and Monaco `DocumentSymobl.detail` is mandatory, in LSP is optional. They are all different APIs.

> The TS-LS fills the detail with an empty string, but according to the LSP, DocumentSymbol.detail is optional. 

TS-LS fills `detail` with empty string to work around a bug in version of `vscode-languageserver-node` which we use for Monaco language client. It can not distinguish between LSP `DocumentSymbol` and `SymbolInformation` otherwise: https://github.com/TypeFox/vscode-languageserver-node/blob/fb80c5e2db4c714032c28d6f482e770642ae9e23/types/src/main.ts#L1701

Reply:
I still don't see why this makes a difference: if there is no value, let's use an emtpy string and be done with. What is the actual problem here?

Reply:
> work around a bug in version of `vscode-languageserver-node` which we use for Monaco language client. It can not distinguish between LSP `DocumentSymbol` and `SymbolInformation`

Note, the issue seems to be fixed in the original source: https://github.com/Microsoft/vscode-languageserver-node/blob/c80ead2e4162cf3f7b547ea2dfc57e64dbd6d423/types/src/main.ts#L1903-L1911